graphql/src/store/prefetch.rs

```diff
@@ -506,38 +466,15 @@ fn execute_root_selection_set(
         None => return Err(vec![QueryExecutionError::NoRootQueryObjectType]),
     };
 
-    // Split the toplevel fields into introspection fields and
-    // 'normal' data fields
-    let mut data_set = q::SelectionSet {
-        span: selection_set.span.clone(),
-        items: Vec::new(),
-    };
-
-    for (_, type_fields) in collect_fields(ctx, &query_type.into(), vec![selection_set], None) {
-        let fields = match type_fields.get(&TypeCondition::Any) {
-            None => return Ok(vec![]),
-            Some(fields) => fields,
-        };
-
-        let name = fields[0].name.clone();
-        let selections = fields
-            .into_iter()
-            .map(|f| q::Selection::Field((*f).clone()));
-        // See if this is an introspection or data field. We don't worry about
-        // nonexistant fields; those will cause an error later when we execute
-        // the query in `execution::execute_root_selection_set`
-        if sast::get_field(query_type, &name).is_some() {
-            data_set.items.extend(selections)
-        }
-    }
```

Comment:
I removed this explicit separation of the introspection fields, because introspection fields will be ignored by field collection anyways.

Reply:
We actually check that in too many places. We have proper validation for `UnknownField` errors in `validate_fields`, so I've removed all other checks for these and replaced them with unwraps.